lib/transformer.js

```diff
@@ -0,0 +1,83 @@
+'use strict'
+
+const aws = require('aws-sdk')
+const zlib = require('zlib')
+
+const formatter = require('./formatter.js')
+
+const BATCH_LIMIT = parseInt(process.env.LOGDNA_BATCH_LIMIT) || 25
+const s3 = new aws.S3({
+  apiVersion: '2006-03-01'
+})
+
+const batchify = (logs) => {
+  const batches = []
+  let batch = [], batch_size = 0
+  logs.forEach((log) => {
+    batch.push(log)
+    batch_size += 1
+    if (batch_size >= BATCH_LIMIT) {
+      batches.push(batch)
+      batch = []
+      batch_size = 0
+    }
+  })
+
+  if (batch_size > 0) { batches.push(batch) }
+  return batches
+}
+
+const extractData = (data) => {
+  const {Records, logFiles, ...meta} = data
+  const ts = meta && meta.eventTime ? (new Date(meta.eventTime)).getTime() : undefined
+  data = Records || logFiles || data
```

Comment:
avoid reassigning the `data` input to this function

```js
let records = data
if (Records) {
  records = Records
} else if (logFiles) {
  records = logFiles
}
```

where does `logFiles` come from, is this a default field from CloudWatch files or something? 

in general I understand the desire to make this s3 handler generic so that it supports all kinds of data that might be flowing through s3, but it comes at the expense of complexity. just something to consider.

Reply:
Done!